recipes/jpeg-compressor/all/test_package/test_package.cpp

```diff
@@ -0,0 +1,38 @@
+#include <iostream>
+
+#include "jpge.h"
+#include "jpgd.h"
+
+int main(int argc, const char* argv[])
+{
+    if (argc < 2) {
+        std::cerr << "Need at least one argument" << std::endl;
+        return 1;
+    }
+
+    const char* pSrc_filename = argv[1];
+
+    const int req_comps = 4;
+    int width = 0, height = 0, actual_comps = 0;
+    uint8_t* pImage_data = jpgd::decompress_jpeg_image_from_file(pSrc_filename, &width, &height, &actual_comps, req_comps);
+    if (!pImage_data)
+    {
+        std::cerr << "Failed loading file " << pSrc_filename << std::endl;
+        return EXIT_FAILURE;
+    }
+
+    int buf_size = width * height * 3;
+    if (buf_size < 1024) buf_size = 1024;
+	void* pBuf = malloc(buf_size);
+    if (!jpge::compress_image_to_jpeg_file_in_memory(pBuf, buf_size, width, height, req_comps, pImage_data))
+    {
+        std::cerr << "Failed creating JPEG data" << std::endl;
+        return EXIT_FAILURE;
+    }
+
+    free(pBuf);
+    // free(pImage_data);
```

Comment:
Should we uncomment this?

Reply:
There is a bug with `Windows x86_64, Debug, Visual Studio 14, MTd. Options: jpeg-compressor:shared-True`. The test_package executable crash when this line is uncommented. I don't know why.

Reply:
Good catch!

Its most likely because the project does not respect the `/MT(d)` flags or it does not support it...

You should uncomment this like and just throw `ConanInvalidConfiguration` for `"MT" in compiler.runtime`

Reply:
Then, please, leave a comment why you commented it.

Reply:
I guess it's buffer overrun. you allocate memory for 3 channels, but specify 4?

Reply:
@prince-chrismc The error only appears with Visual Studio And Debug And MTd And Shared (`-s build_type=Debug -s compiler.runtime=Visual Studio -s compiler.runtime=MTd -o jpeg-compressor:shared=True`). It works with Release and MT.